friends/src/reducers/index.js

```diff
@@ -0,0 +1,72 @@
+
+import {
+    LOGIN_SUCCESS,
+    LOGIN_FAILURE,
+    LOGIN_START,
+    FETCH_FRIENDS_START,
+    FETCH_FRIENDS_SUCCESS,
+    FETCH_FRIENDS_FAILURE,
+    DELETE_FRIEND
+  } from "../actions";
+  
+
+const initialState = {
+    deletingFriend: false,
+    fetchingFriends: false,
+    friends: [],
+    loggingIn: false,
+    savingFriends: false,
+    updatingFriend: false,
+    error: null
+    // Array characters, Boolean fetching, null error.
+  };
+
+
+
+  export const rootReducer =(state=initialState, action) => {
+      switch(action.type) {
+          case LOGIN_START:
+            return {
+                ...state,
+                loggingIn: true
+            }
+            case LOGIN_SUCCESS:
+                return {
+                ...state,
+                loggingIn: false
+            }
+            case LOGIN_FAILURE: 
+            return {
+                ...state,
+                loggingIn: false,
+                error: action.payload
+            }
+            case FETCH_FRIENDS_START:
+                return {
+                    ...state,
+                    fetchingFriends: true,
+                    error: ''
```

Comment:
When there is no error, keep it set to `null` instead of an empty string `""` - `null` states more declaratively that there is no error, as opposed to having an 'error' that is composes of empty text.